contracts/reputation/Coreputation.sol

```diff
@@ -74,6 +74,47 @@ contract Coreputation is MasterCopyNonUpgradable, CoConsensusModule {
         CoConsensusModule.setupCoConsensus(_coConsensus);
     }
 
+    /**
+     * @notice Insert or update validator information.
+     *
+     * @dev Validator status is marked Staked, if validator status is
+     *      Undefined and he has non zero reputation value.
+     *      Validator is not inserted if his reputation value is zero.
+     *      Validator status is marked to Deregistered when his current status
+     *      is Staked and reputation value changes to zero.
+     *
+     * @dev Function requires:
+     *      - msg.sender should be CoConsensus
+     *      - Validator address should non zero
+     *      - Newly added validators reputation should be non zero
+     *
+     * @param _validator Validator address to upsert.
+     * @param _reputation Validator reputation value.
+     */
+    function upsertValidator(
+        address _validator,
+        uint256 _reputation
+    )
+        external
+        onlyCoConsensus
+    {
+        ValidatorInfo storage vInfo = validators[_validator];
+        if (vInfo.status == ValidatorStatus.Undefined) {
+            if (_reputation == uint256(0)) {
+                return;
+            }
+
+            vInfo.status = ValidatorStatus.Staked;
+            vInfo.reputation = _reputation;
+        } else {
+            if(_reputation == uint256(0) &&
+                vInfo.status == ValidatorStatus.Staked) {
+                vInfo.status = ValidatorStatus.Deregistered;
+                vInfo.reputation = _reputation;
```

Comment:
I think that the reputation should always be updated so that it will be in sync with what is received from the origin chain. Only the validator status changes should be conditional.

Reply:
Should it be updated when validator is slashed?

Reply:
no, `Slashed` should always remain `Slashed`